client/src/redux/reducers/location.ts

```diff
@@ -0,0 +1,155 @@
+import type { LocationInfo, WeatherInfo } from 'types';
+import { ActionType, Action } from '../actionTypes/location';
+
+interface LocationState {
+  search: {
+    data: LocationInfo[] | null;
+    loading: boolean;
+    error: string | null;
+  };
+  info: {
+    data: LocationInfo | null;
+    loading: boolean;
+    error: string | null;
+  };
+  weather: {
+    current: {
+      data: WeatherInfo | null;
+      loading: boolean;
+      error: string | null;
+    };
+  };
+  query: string;
+}
+
+const initialState = {
+  search: {
+    data: null,
+    loading: false,
+    error: null
+  },
+  info: {
+    data: null,
+    loading: false,
+    error: null
+  },
+  weather: {
+    current: {
```

Comment:
Do you need use 'current'. I think you can save weather data without 'current'
`weather: {
  data: ...
}`